compiler/rustc_lint/src/context/diagnostics.rs

```diff
@@ -105,7 +105,7 @@ pub(super) fn builtin(sess: &Session, diagnostic: BuiltinLintDiag, diag: &mut Di
         BuiltinLintDiag::RedundantImport(spans, ident) => {
             for (span, is_imported) in spans {
                 let introduced = if is_imported { "imported" } else { "defined" };
-                let span_msg = if span.is_dummy() { "by prelude" } else { "here" };
+                let span_msg = if span.is_dummy() { "by the extern prelude" } else { "here" };
```

Comment:
actually it might be possible that we have a dummy span if the source code of the std library is not available

Reply:
Ah nvm, that's not the case :facepalm: If the source is not available, the diagnostic just ends up being butchered (tested with `beta`, rustup profile `minimal`):

```
warning: the item `TryInto` is imported redundantly
 --> z.rs:1:5
  |
1 | use std::convert::TryInto;
  |     ^^^^^^^^^^^^^^^^^^^^^
 --> /rustc/efd9d2df12b5e17fac0b4d0fb89f612ecd79f259/library/std/src/prelude/mod.rs:129:13
  |
  = note: the item `TryInto` is already defined here
  |
  = note: `#[warn(unused_imports)]` on by default

warning: 1 warning emitted
```

but the `Span` is unaffected ofc.